From 53d9e933bc78c7d28b92a7e723ee1b0cd00d69d7 Mon Sep 17 00:00:00 2001
From: Huibin Hong <huibin.hong@rock-chips.com>
Date: Thu, 8 Feb 2018 11:29:38 +0800
Subject: [PATCH] fiq_debugger: rockchip: deal with this_cpu and current_cpu
 are different

If this cpu is not the same as current cpu, it means we don't use
current cpu any more, and this cpu handles uart interrupt. So it
is unnecessary to set uart interrupt to this cpu again.

Change-Id: Ia454af96ee766dcbc8b22339652a4aa7eb229568
Signed-off-by: Huibin Hong <huibin.hong@rock-chips.com>
---
 drivers/staging/android/fiq_debugger/fiq_debugger.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/staging/android/fiq_debugger/fiq_debugger.c b/drivers/staging/android/fiq_debugger/fiq_debugger.c
index be7ef2512610..98bfb26bbcea 100644
--- a/drivers/staging/android/fiq_debugger/fiq_debugger.c
+++ b/drivers/staging/android/fiq_debugger/fiq_debugger.c
@@ -866,8 +866,7 @@ static bool fiq_debugger_handle_uart_interrupt(struct fiq_debugger_state *state,
 	int count = 0;
 	bool signal_helper = false;
 
-	if ((this_cpu != state->current_cpu) &&
-	    (cpu_online(state->current_cpu))) {
+	if (this_cpu != state->current_cpu) {
 		if (state->in_fiq)
 			return false;
 
@@ -881,13 +880,10 @@ static bool fiq_debugger_handle_uart_interrupt(struct fiq_debugger_state *state,
 			this_cpu);
 
 		atomic_set(&state->unhandled_fiq_count, 0);
-		fiq_debugger_switch_cpu(state, this_cpu);
+		state->current_cpu = this_cpu;
 		return false;
 	}
 
-	if (this_cpu != state->current_cpu)
-		state->current_cpu = this_cpu;
-
 	state->in_fiq = true;
 
 	while ((c = fiq_debugger_getc(state)) != FIQ_DEBUGGER_NO_CHAR) {
-- 
2.35.3

